app/lib/zeebe-api.js

```diff
@@ -0,0 +1,353 @@
+
+    const client = this._getZeebeClient(endpoint);
+
+    try {
+      const response = await client.deployWorkflow({
+        definition: contents,
+        name: prepareDeploymentName(name, filePath)
+      });
+
+      return {
+        success: true,
+        response: response
+      };
+    } catch (err) {
+      log.error('Failed to deploy with config (secrets omitted):', withoutSecrets(parameters), err);
+
+      return {
+        success: false,
+        response: err
+      };
+    }
+  }
+
+  /**
+   * @public
+   * Run process instance.
+   *
+   * @param {ZeebeClientParameters & { processId: string }} parameters
+   * @returns {{ success: boolean, response: object }}
+   */
+  async run(parameters) {
+
+    const {
+      endpoint,
+      processId
+    } = parameters;
+
+    const client = this._getZeebeClient(endpoint);
+
+    try {
+
+      const response = await client.createWorkflowInstance({
+        bpmnProcessId: processId
+      });
+
+      return {
+        success: true,
+        response: response
+      };
+    } catch (err) {
+      log.error('Failed to run instance with config (secrets omitted):', withoutSecrets(parameters), err);
+
+      return {
+        success: false,
+        response: err
+      };
+    }
+  }
+
+  _getCachedZeebeClient(endpoint) {
+    const cachedEndpoint = this._cachedEndpoint;
+
+    if (isHashEqual(endpoint, cachedEndpoint)) {
+      return this._zeebeClient;
+    }
+  }
+
+  _getZeebeClient(endpoint) {
+
+    // (1) use existing Zeebe Client for endpoint
+    const cachedZeebeClient = this._getCachedZeebeClient(endpoint);
+
+    if (cachedZeebeClient) {
+      return cachedZeebeClient;
+    }
+
+    // (2) cleanup old client instance
+    this._shutdownZeebeClientInstance();
+
+    // (3) create new Zeebe Client for endpoint configuration
+    this._zeebeClient = this._createZeebeClient(endpoint);
+    this._cachedEndpoint = endpoint;
+
+    return this._zeebeClient;
+  }
+
+  _shutdownZeebeClientInstance() {
+    this._zeebeClient && this._zeebeClient.close();
+  }
+
+  _createZeebeClient(endpoint) {
+    const {
+      type,
+      url
+    } = endpoint;
+
+    let options = {
+      retry: false
+    };
+
+    if (!values(endpointTypes).includes(type)) {
+      return;
+    }
+
+    if (type === endpointTypes.OAUTH) {
+      options = {
+        ...options,
+        oAuth: {
+          url: endpoint.oauthURL,
+          audience: endpoint.audience,
+          clientId: endpoint.clientId,
+          clientSecret: endpoint.clientSecret,
+          cacheOnDisk: false
+        },
+        useTLS: true
+      };
+    }
+
+    if (type === endpointTypes.CAMUNDA_CLOUD) {
+      options = {
+        ...options,
+        camundaCloud: {
+          clientId: endpoint.clientId,
+          clientSecret: endpoint.clientSecret,
+          clusterId: endpoint.clusterId,
+          cacheOnDisk: false
+        },
+        useTLS: true
+      };
+    }
```

Comment:
These two conditions are mutually exclusive, so I'd suggest to merge them with `else if` so that the code is easier to follow.

Reply:
👍